pkg/util/k8s/node.go

```diff
@@ -51,16 +51,14 @@ func GetNodeAddrsWithType(node *v1.Node, types []v1.NodeAddressType) (*ip.DualSt
 	}
 
 	nodeAddrs := new(ip.DualStackIPs)
-	for i := range ipAddrStrs {
-		addr := net.ParseIP(ipAddrStrs[i])
-		if addr == nil {
-			return nil, fmt.Errorf("'%s' is not a valid IP address", ipAddrStrs[i])
-		}
-		if addr.To4() == nil {
-			nodeAddrs.IPv6 = addr
-		} else {
-			nodeAddrs.IPv4 = addr
-		}
+	addr := net.ParseIP(ipAddrStrs[0])
+	if addr == nil {
+		return nil, fmt.Errorf("'%s' is not a valid IP address", ipAddrStrs[0])
+	}
+	if addr.To4() == nil {
+		nodeAddrs.IPv6 = addr
+	} else {
+		nodeAddrs.IPv4 = addr
```

Comment:
By first IP, I meant the first IP of each family in the list. There could still be other type of addresses like InternalDNS and Hostname, and there could be one IPv4 and IPv6 address at the same time in dual-stack cluster.

Reply:
Updated to fix the case for dual-stack IPs. Thanks!